src/controllers/user.controller.ts

```diff
@@ -17,13 +23,21 @@ export const createUser = async (req: Request, res: Response) => {
   }
 };
 
+/**
+ * Retrieves a user's data based on their userId and returns it as an object.
+ * @param {Request} req - Express request object containing the query parameters, including the userId.
+ * @param {Response} res - Express response object used to send the response back to the client.
+ * @returns {Object} Returns status code 200 and an object containing a success message and the targetUser object if successful.
+ * Otherwise, returns an error status with an appropriate message.
+ */
 export const getUser = async (req: Request, res: Response) => {
-  const userId = req.query?.userId!.toString();
-  try {
-    if (!ObjectId.isValid(userId)) {
-      return res.status(400).send({ message: 'Invalid user ID!' });
-    }
+  const userId = req.query?.userId?.toString() || '';
```

Comment:
Instead of `|| ''`, why don't you check if `userId` is undefined and return a bad request response saying no `userId` was given? I think that is better handling :)